common/src/main/java/com/wynntils/wynn/model/ShamanTotemModel.java

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Model;
+import com.wynntils.mc.event.AddEntityEvent;
+import com.wynntils.mc.event.ChangeCarriedItemEvent;
+import com.wynntils.mc.event.RemoveEntitiesEvent;
+import com.wynntils.mc.event.SetEntityDataEvent;
+import com.wynntils.mc.objects.Location;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.utils.Delay;
+import com.wynntils.wynn.event.CharacterUpdateEvent;
+import com.wynntils.wynn.event.SpellCastedEvent;
+import com.wynntils.wynn.event.TotemActivatedEvent;
+import com.wynntils.wynn.event.TotemRemovedEvent;
+import com.wynntils.wynn.event.TotemSummonedEvent;
+import com.wynntils.wynn.objects.ShamanTotem;
+import com.wynntils.wynn.objects.SpellType;
+import com.wynntils.wynn.utils.WynnUtils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.syncher.SynchedEntityData;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.decoration.ArmorStand;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.phys.AABB;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class ShamanTotemModel extends Model {
+
+    private ShamanTotem totem1 = null;
+    private Integer pendingTotem1Id = null;
+
+    private ShamanTotem totem2 = null;
+    private Integer pendingTotem2Id = null;
+
+    private ShamanTotem totem3 = null;
+    private Integer pendingTotem3Id = null;
+
+    private long totemCastTimestamp = 0;
+    private int nextTotemSlot = 1;
+    private int summonWeaponSlot = -1;
+
+    private static final Pattern SHAMAN_TOTEM_TIMER = Pattern.compile("§c(\\d+)s");
+
+    @SubscribeEvent
+    public void onTotemSpellCast(SpellCastedEvent e) {
+        if (e.getSpell() != SpellType.TOTEM) return;
+
+        totemCastTimestamp = System.currentTimeMillis();
+        summonWeaponSlot = McUtils.inventory().selected;
+    }
+
+    @SubscribeEvent
+    public void onTotemSpawn(AddEntityEvent e) {
+        Delay.create(
```

Comment:
If your goal was to run on next tick, I would use `MinecraftSchedulerManager`. It is far less costly, and we will be doing this a lot.

Reply:
(Delay 1 and `MinecraftSchedulerManager` does the same thing, I think, bug the manager should be much faster in theory.)